src/systems/logical_audio_sensor_plugin/Source.hh

```diff
@@ -0,0 +1,245 @@
+                   const float _falloffDistance,
+                   const float _volumeLevel,
+                   const bool _playing);
+
+    /// \brief Computes the volume level of the source at a specific location.
+    /// \param[in] _location The location where the volume level should be
+    ///   calculated.
+    /// \return The volume level at this location.
+    ///   If the attenuation function or shape is undefined, -1.0 is returned.
+    ///   If the source is not playing, 0.0 is returned.
+    public: float VolumeLevel(const ignition::math::Vector3d &_location) const;
+
+    /// \brief Start playing audio from the source. If the source is already
+    ///   playing audio, nothing will change (the source will keep playing).
+    public: void StartPlaying();
+
+    /// \brief Stop playing audio from the source. If the source has already
+    ///   been stopped, nothing will change (the source will remain stopped).
+    public: void StopPlaying();
+
+    /// \brief Check to see if the audio source is playing or not.
+    /// \return \c true if the source is playing, \c false otherwise.
+    public: bool IsPlaying() const;
+
+    /// \brief Set the source's position.
+    /// \param[in] _position The new position of the source.
+    public: void SetPosition(const ignition::math::Vector3d &_position);
+
+    /// \brief Set the attenuation function that matches the defined string.
+    ///   The string is case sensitive, and must match the spelling
+    ///   of the values in AttenuationFunction. If the spelling does not match,
+    ///   the attenuation function is set as Undefined.
+    ///
+    ///   \em Example: to set the attenuation function to
+    ///     AttenuationFunction::Linear, the following
+    ///     are examples of valid strings: "linear", "Linear", "LINEAR"
+    /// \param[in] _attenuationFunc a string that should map to a value in
+    ///   AttenuationFunction.
+    private: void SetAttenuationFunction(std::string _attenuationFunc);
+
+    /// \brief Set the attenuation shape that matches the defined string.
+    ///   The string is case sensitive, and must match the spelling
+    ///   of the values in AttenuationShape. If the spelling does not match,
+    ///   the attenuation shape is set as Undefined.
+    ///
+    ///   \em Example: to set the attenuation shape to
+    ///     AttenuationShape::Sphere, the following
+    ///     are examples of valid strings: "sphere", "Sphere", "SPHERE"
+    /// \param[in] _attenuationShape a string that should map to a value in
+    ///   AttenuationShape.
+    private: void SetAttenuationShape(std::string _attenuationShape);
+
+    /// \brief Set the inner radius for the audio source.
+    /// \param[in] _innerRadius The inner radius to set for the source.
+    ///   This value must be > 0.
+    ///   If the value of this parameter is < 0, the source's inner radius
+    ///   will be set to 0.
+    private: void SetInnerRadius(const float _innerRadius);
```

Comment:
`const` isn't really necessary on primitives passed by value.  While completely valid, it doesn't really have an impact.

Reply:
The main reason why I marked the primitives as `const` was to protect myself from accidentally modifying any values that didn't need to be changed in a method. Should I remove the `const` qualifier for the primitives?